BSP/STM32F413/Src/BSP_GPIO.c

```diff
@@ -0,0 +1,102 @@
+/* Copyright (c) 2021 UT Longhorn Racing Solar */
+
+#include "BSP_GPIO.h"
+#include "stm32f4xx.h"
+#include "Tasks.h"
+
+OS_SEM GPIO_Update_Sem4;
+
+/**
+ * @brief   Initializes a GPIO port
+ * @param   port -port to initialize, mask - pins, write - input or output
+ * @return  None
+ */ 
+void BSP_GPIO_Init(port_t port, uint16_t mask, uint8_t write){
+	
+	GPIO_InitTypeDef GPIO_InitStruct;
+
+	RCC_AHB1PeriphClockCmd(1 << port, ENABLE);
+	// Configure the pins to be generic GPIO
+    GPIO_InitStruct.GPIO_Pin   = mask;
+    GPIO_InitStruct.GPIO_Mode  = write ? GPIO_Mode_OUT : GPIO_Mode_IN;
+    GPIO_InitStruct.GPIO_OType = GPIO_OType_PP;
+    GPIO_InitStruct.GPIO_Speed = GPIO_Low_Speed;
+    GPIO_InitStruct.GPIO_PuPd  = GPIO_PuPd_NOPULL;  // TODO: verify
+
+    // Compute the offset for the port handle from the port passed in
+    GPIO_TypeDef *portHandle = GET_HANDLE_FROM_ENUM(port);

```

Comment:
Where is the definition for this `GET_HANDLE_FROM_ENUM` macro? It seems like you could use the `GPIO_GetPort` function here instead.

Reply:
Yep, I'll change it to use GPIO_GetPort, GET_HANDLE_FROM_ENUM was a macro we could use instead of the function but we sticked to using the function